### DISCUSSION TITLE
同步助手 · 命令行版本

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3704] [同步助手]
![Sync CLI](https://user-images.githubusercontent.com/81074/163512494-3c41c6c3-2158-49f8-9425-27637a3303c8.png)

> 来自 [简悦社区](https://t.me/simpread) 用户 [Jigsaw](https://github.com/j1g5awi) 的 Golang 版本，实现了大部分同步助手的功能。

# 描述

同步助手的命令行版本，可以替代部分同步助手的功能。

# 下载地址

https://github.com/j1g5awi/simpread-sync/releases

# 功能比较

| 功能                                 | 同步助手          | 命令行                   | 备注      |
|------------------------------------|---------------|-----------------------|---------|
| 自动同步                               | ●             | ●                     | -       |
| HTML、Markdown 导出                   | ●             | ●                     | -       |
| Email                              | ●             | ●                     | -       |
| Epub                               | ●             | ●                     | -       |
| Texbundle                          | ●             | ●                     | -       |
| PDF                                | ●             | ○                     | 客户端独有功能 |
| 内置解析                               | ●             | ○                     | 客户端独有功能 |
| 小书签                                | ●             | ○                     | 客户端独有功能 |
| 标注的自动同步（Hypothes.is / Readwise.io） | ●             | ○                     | 客户端独有功能 |
| 快照                                 | ●             | ●                     | -       |
| 客户端                                | Mac / Windows | Mac / Windows / Linux | -       |

# 使用

```sh
./simpread-sync -c <configPath>
```

`config.json` 默认配置如下：

```json
{
    "port": 7026,
    "syncPath": "",
    "outputPath": "",
    "smtpHost": "",
    "smtpPort": 465,
    "smtpUsername": "",
    "smtpPassword": "",
    "mailTitle": "[简悦] - {{ title }}",
    "receiverMail": "",
    "kindleMail": ""
}
```

备注：

1. `syncPath` 必须填写，否则无法自动同步。
2. `outputPath` 如果不填写，默认为 `syncPath` 下的 `output` 文件夹。

# 如何更新

使用 `./simpread-sync -V` 来检查当前版本。（如有更新则会自动提示）

![image](https://user-images.githubusercontent.com/81074/162721397-ba796bc7-2d5a-4bd7-8472-f1aa7bfb3be0.png)

# 一个例子

1. 根据下载地址下载对应的系统版本，并解压缩到任意目录

    ![image](https://user-images.githubusercontent.com/81074/162691202-d27ae581-d76c-494a-8676-ee43981c21bf.png)

2. 并输入如下配置信息。

```
{
    "port": 7026,
    "syncPath": "***/Nutstore/SimpRead",
    "outputPath": "***/Nutstore/SimpRead/output",
    "mailTitle": "[简悦] - {{title}}",
    "smtpHost": "smtp.qq.com",
    "smtpUsername": "***@qq.com",
    "smtpPassword": "***",
    "receiverMail": "***@qq.com",
    "kindleMail": "***@kindle.cn"
}
```

3. 在命令行下输入 `./simpread-sync` 即可正常运行。（如下图所示）

    ![image](https://user-images.githubusercontent.com/81074/162691582-987c35c0-27c6-4773-acd2-6d3f42d85b69.png)

# 适用范围

1. 希望得到一个更轻量级的使用方案的用户

2. Linux 用户

3. 因 AMD 显卡出现错误的用户

# 注意事项

1. 与同步助手一样，此功能为高级账户功能。
2. 此功能建议跟 **守护进程** 一起使用，暂时没有这方面的教程（正式版发布后会补上），一般来说 Linux 用户可直接使用 `systemd` 方式。

# 一些补充说明

https://github.com/Kenshin/simpread/discussions/3704#discussioncomment-2543727

# 问题反馈

https://github.com/Kenshin/simpread/issues/3740

# 版本性质

1. 虽然代码来自 @j1g5awi 但全部经过简悦官方审核，所以纯粹的社区版本，而是官方协作下的社区版。

2. 下载位置在 @j1g5awi 的库，是为了尊重开发者，所以没有转移到简悦官方 repo 里面。

3. 此版本会随着简悦同步助手的版本升级而已升级，但并不是每个功能都一一对应，我会把一些轻量级的功能已知到命令行版本。

4. 仍会有 Linux Desktop 版的同步助手。

谢谢 @ihipop 的提示。

# 关联问题

- https://github.com/Kenshin/simpread/issues/1450
- https://github.com/Kenshin/simpread/issues/2293
- https://github.com/Kenshin/simpread/issues/2793
- https://github.com/Kenshin/simpread/issues/3297

**Comment by j1g5awi** [comment]
# 一些补充说明

`simpread-sync` 支持 UNIX 式的命令行调用，如果你只是想要指定同步的文件夹，可以直接 ：

`./simpread-sync --sync-path <syncPath>`

以及你还可以把 markdown 文件和 textBundle 文件导出到另一个指定的文件夹：

`./simpread-sync --sync-path <syncPath> --markdown-path <markdownPath>`

更多用法请使用 `./simpread-sync -h` 查看。

另外，如果你是 Linux 用户，我在主分支留下了 sytemed 的 service 模板；Arch Linux 用户还可以在该仓库的 `pkgbuild` 分支发现我提供的 PKGBUILD 文件。

如果你是 scoop 用户，也欢迎添加我的 scoop bucket https://github.com/j1g5awi/scoop-jigsaw 来安装此程序。
